### PR TITLE
[CLI-22] Add `op version list --project` command

### PR DESCRIPTION
**Pull request by cbliard**
# Ticket

https://community.openproject.org/wp/CLI-22

# What are you trying to accomplish?

The `--version`/`--not-version` work-package filters accept version IDs, but there was no way to discover them. This commit adds a command to list versions of a project for this purpose.

## Screenshots

image

# What approach did you choose and why?

Actually it adds the command back: the absence of version listing was a regression from commit 17de261, which originally resolved `--version` by name and printed the available-versions list as a fallback; that lookup path (and its only caller) was dropped in bbef582 when the filter switched to raw IDs.

So it rewires the dead code back, and use the printer `Renderer` interface to have both text and json output available.

# Merge checklist

- [x] Added/updated tests
- [x] Tested live with OpenProject 17.7 and 17.8

### CODE REVIEW

**Comment by myabc** [comment]
@cbliard I'd personally find this more useful if we could sort the items (locally?) before displaying.

File: cmd/version/list_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	err := listVersions(nil, nil)
+	if !errors.Is(err, openerrors.ErrHandled) {
+		t.Fatalf("listVersions error = %v, want ErrHandled", err)
+	}
+	if count := strings.Count(testingPrinter.ErrResult, "[ERROR]"); count != 1 {
+		t.Errorf("error diagnostic count = %d, want 1; stderr: %q", count, testingPrinter.ErrResult)
+	}
+}
+
+func TestListVersionsPrintsAvailableVersions(t *testing.T) {
```
Comment (myabc) [inline cmd/version/list_test.go]: Although an unlikely scenario in real-world usage (at least aT OP), We're missing a test for an empty version collection. This contrasts with `cmd/workpackage/search.go:51` which explicitly prints `No work package found for search input`.

File: components/printer/versions_test.go [hunk 2]
```diff
@@ -14,27 +12,14 @@ func TestVersions(t *testing.T) {
...
-	expected := common.Reduce(
-		versions,
-		func(state string, version *models.Version) string {
-			idString := "#" + strconv.FormatUint(version.Id, 10)
-			return state + fmt.Sprintf("[%s] %s\n", printer.Red(idString), printer.Cyan(version.Name))
-		},
-		"")
+	expected := fmt.Sprintf("%s %s\n", printer.Red(" #2"), printer.Cyan("13.0"))
+	expected += fmt.Sprintf("%s %s\n", printer.Red(" #4"), printer.Cyan("13.1"))
+	expected += fmt.Sprintf("%s %s\n", printer.Red("#43"), printer.Cyan("45.5"))
```
Comment (myabc) [inline components/printer/versions_test.go]: We're only testing `TextRenderer`. Shouldn't we exercise `JsonRenderer` as well?

File: cmd/version/list.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+var listCmd = &cobra.Command{
+	Use:   "list",
+	Short: "Lists versions",
+	Long:  "Get a list of a project's versions, scoped by the provided flag (--project).",
+	RunE:  listVersions,
+}
+
+func listVersions(_ *cobra.Command, _ []string) error {
+	if err := projects.ValidateIdentifier(listProjectId); err != nil {
+		printer.ErrorText(err.Error())
```
Comment (myabc) [inline cmd/version/list.go]: Shouldn't we contextualise the error message? (as `cmd/budget/list.go:25` does)

```suggestion
		printer.ErrorText(fmt.Sprintf("--project: %s", err.Error()))
```

File: cmd/root.go [hunk 4]
```diff
@@ -17,6 +17,7 @@ import (
 	"github.com/opf/openproject-cli/cmd/status"
 	"github.com/opf/openproject-cli/cmd/timeentry"
 	"github.com/opf/openproject-cli/cmd/user"
+	"github.com/opf/openproject-cli/cmd/version"
```
Comment (myabc) [inline cmd/root.go]: **Import shadowing** (claude finding): `cmd/root.go:143`'s `func Execute(version *configuration.Version)` parameter strictly shadows the newly added `github.com/opf/openproject-cli/cmd/version` import.

**Review by myabc — CHANGES_REQUESTED** [review]
Looks good overall. Just a few issues:

**Naming the subcommand**

Many popular CLIs support a version subcommand and flag, e.g. GitHub'S CLI which supports `gh version` and `gh --version`. On the other hand, `op version` and `op --version` do different things. Are we ok with the (conceptual) collision here?

Noun first has made this more confusing IMO.

The `--version `flag also collides/means different things: `op --version` vs `work-package list --project DREAM --version 2323`.

_I don't really have a better suggestion. Renaming the noun to something else (e.g. `project-versions` or `releases` would probably be even more confusing/problematic for users already familiar with OpenProject. Just wanted to highlight this!_

**Documentation**

`CLAUDE.md` hasn't been updated to mention the subcommand.

**Other issues**

Claude found two additional issues with existing code, that may now become apparent:

* **Nil pointer panic:**: `dtos/version.go:33` directly dereferences `dto.Embedded.Elements` without a prior `if dto.Embedded == nil guard` (unlike `dtos/budget.go:21`).
* **Silent pagination truncation**: `components/resources/projects/versions.go:12` passes `nil` instead of `requests.NewPaginatedQuery(-1, nil)` for the query, leaving it at default page limits.

File: cmd/version/list_test.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	server := httptest.NewServer(http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+		response.Header().Set("Content-Type", "application/json")
+		_, _ = response.Write([]byte(`{
```
[the remark this answers is not in the document]
Comment (myabc) [reply cmd/version/list_test.go]: Not sure about this finding.

File: cmd/version/list_test.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	t.Cleanup(func() { listProjectId = "" })
+
+	if err := listVersions(nil, nil); err != nil {
+		t.Fatalf("listVersions error = %v, want nil", err)
+	}
+
+	if !strings.Contains(testingPrinter.Result, "v17") || !strings.Contains(testingPrinter.Result, "v42") {
+		t.Errorf("expected output to contain versions 'v17' and 'v42', got: %q", testingPrinter.Result)
+	}
+}
```
[the remark this answers is not in the document]
Comment (myabc) [reply cmd/version/list_test.go]: I think this is the same as https://github.com/opf/openproject-cli/pull/21#discussion_r3805904424

File: cmd/version/list_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (cbliard) [reply cmd/version/list_test.go]: Same. It was there initially, but I removed it thinking that it's just a stub server. Now I realize that there may be value in testing the verb and http endpoint.

**Comment by cbliard** [comment]
> @cbliard I'd personally find this more useful if we could sort the items (locally?) before displaying.

In a way `| sort` cannot handle?
